src/object_storage/objects.rs

```diff
@@ -234,6 +248,49 @@ impl IntoFallibleIterator for ObjectQuery {
     }
 }
 
+impl<R: Read + Send + 'static> NewObject<R> {
+    /// Start creating an object.
+    pub(crate) fn new(session: Rc<Session>, c_name: String, name: String, body: R) -> NewObject<R> {
+        NewObject {
+            session,
+            name,
+            c_name,
+            body,
+            delete_after: None,
+            delete_at: None,
+        }
+    }
+
+    /// Request creation of the object.
+    pub fn create(self) -> Result<Object> {
+        let c_name = self.c_name.clone();
+        let object_create = protocol::ObjectCreate {
+            body: self.body,
+            c_name: self.c_name,
+            name: self.name,
+            delete_after: self.delete_after,
+            delete_at: self.delete_at,
+        };
+
+        let inner = api::create_object(&self.session, object_create)?;
+        Ok(Object::new(self.session, inner, c_name))
+    }
+
+    /// Set ttl for object
+    #[inline]
+    pub fn with_delete_after(mut self, ttl: u32) -> NewObject<R> {
+        self.delete_after = Option::from(ttl);
+        self
+    }
+
+    /// Set the datetime when the object must be deleted
+    #[inline]
+    pub fn with_delete_at(mut self, timestamp: i64) -> NewObject<R> {
```

Comment:
To make the API more rusty, I'd prefer to avoid naked timestamps here in the high-level API. What if we use https://docs.rs/chrono/0.4.10/chrono/struct.DateTime.html instead? From that you can easily get a timestamp.

Reply:
I thought about it but I didn't know if chrono was a widely used lib. So I didn't use it to not force people to add the dependency.

More I read, more I found that chrono is a kind of a standard lib.

I modified `with_delete_at` to use a `DateTime` in 57ba9ce03c6ff6760699cdb7f267fbad0b061f94